plenum/server/replica.py

```diff
@@ -1122,6 +1122,12 @@ def report_suspicious(reason):
                              "then the latest one - ignoring it".format(key))
         elif why_not == PP_CHECK_REQUEST_NOT_FINALIZED:
             non_fin_reqs = self.nonFinalisedReqs(pre_prepare.reqIdr)
+            for req in non_fin_reqs:
+                if req not in self.requests and self.node.seqNoDB.get(req) != (None, None):
+                    self.logger.info("Request digest {} already ordered. Discard {} "
+                                     "from {}".format(req, pre_prepare, sender))
+                    report_suspicious(Suspicions.DUPLICATE_PPR_SENT)
```

Comment:
Should it be DUPLICATE_PPR_SENT or a new suspicions PPR_WITH_ORDERED_REQUEST?

Reply:
I think PPR_WITH_ORDERED_REQUEST is better

Reply:
Done